### ISSUE TITLE
yarn install kills the build

### ISSUE THREAD

**Issue by carukc** [sameersbn/docker-gitlab#1199]
When bash is set to die on any error (set -e), install.sh will not complete due to a problem installing yarn:
```
yarn install v0.23.2
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.17: The platform "linux" is incompatible with this module.
info "fsevents@1.0.17" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 16.41s.
Compiling assets. Please be patient, this could take a while...
The command '/bin/sh -c bash ${GITLAB_BUILD_DIR}/install.sh' returned a non-zero code: 1
Makefile:25: recipe for target 'build' failed
make: *** [build] Error 1
```
changing 'set -e' in install.sh to 'set -x' allows the build to finish but I'm not sure if the missing dependency is going to cause a problem.
```
yarn install v0.23.2
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.17: The platform "linux" is incompatible with this module.
info "fsevents@1.0.17" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 16.07s.
```

It's my understanding that yarn just reduces the time it take to install by caching packages and parallelizeing package installs so not being there should not cause any problems.

Has anyone out there experienced any issues?

Chris

**Comment by solidnerd** [comment]
Hey,
i have the same error on some cloud machines. It runs on my normal machines quite well but if build it on a digitalocean instance i got the same problem. I'm interessted in why yarn kills it.

**Comment by carukc** [comment]
Hi, I've not figured it out yet but in my case, it's a local system running Ubuntu 16.10 and latest docker but the build itself uses the Dockerfile that's in this repo.

Since the whole point of Docker is to create a repeatable build it's hard to see how it's possible that it should appear some time but not others...  perhaps the repo that we happen to be pulling files from?

**Comment by solidnerd** [comment]
Could you post me your kernel version you are running on this is the only thing that is shared.

**Comment by carukc** [comment]
I'm running 4.8.0-46-generic

**Comment by solidnerd** [comment]
I'm still confused. It's running on travis but If i do it on one of my gitlab runners have the same problem again.

**Comment by carukc** [comment]
Can you tell have version of yarn loads when it does work?   I've tried the build on several systems. Always the same result. I'll try on a copy of 16.04 tomorrow.

**Comment by solidnerd** [comment]
It seems that yarn it isn't. Here is a public build on travis and it works. https://travis-ci.org/sameersbn/docker-gitlab/builds/227655243?utm_source=github_status&utm_medium=notification .  I added tried to get logging from this and here is my output. 

```
yarn install v0.23.3
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.17: The platform "linux" is incompatible with this module.
info "fsevents@1.0.17" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 21.78s.
Last Exit Code: 0
Compiling assets. Please be patient, this could take a while...
Missing Rails.application.secrets.secret_key_base for production environment. The secret will be generated and stored in config/secrets.yml.
Missing Rails.application.secrets.otp_key_base for production environment. The secret will be generated and stored in config/secrets.yml.
Missing Rails.application.secrets.db_key_base for production environment. The secret will be generated and stored in config/secrets.yml.
Missing Rails.application.secrets.jws_private_key for production environment. The secret will be generated and stored in config/secrets.yml.
rake aborted!
Errno::EADDRNOTAVAIL: Cannot assign requested address - connect(2) for [::1]:6379
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:122:in `connect_addrinfo'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:162:in `block in connect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:160:in `each'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:160:in `each_with_index'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:160:in `connect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/connection/ruby.rb:211:in `connect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:323:in `establish_connection'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:94:in `block in connect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:280:in `with_reconnect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:93:in `connect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:351:in `ensure_connected'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:208:in `block in process'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:293:in `logging'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:207:in `process'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:178:in `call_pipelined'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:150:in `block in call_pipeline'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:280:in `with_reconnect'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis/client.rb:148:in `call_pipeline'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis.rb:2199:in `block in pipelined'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis.rb:57:in `block in synchronize'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis.rb:57:in `synchronize'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-3.2.2/lib/redis.rb:2195:in `pipelined'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-namespace-1.5.2/lib/redis/namespace.rb:414:in `namespaced_block'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/redis-namespace-1.5.2/lib/redis/namespace.rb:266:in `pipelined'
/home/git/gitlab/config/initializers/sidekiq.rb:71:in `block in <top (required)>'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/sidekiq-4.2.7/lib/sidekiq.rb:95:in `block in redis'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:64:in `block (2 levels) in with'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:63:in `handle_interrupt'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:63:in `block in with'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:60:in `handle_interrupt'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:60:in `with'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/sidekiq-4.2.7/lib/sidekiq.rb:92:in `redis'
/home/git/gitlab/config/initializers/sidekiq.rb:70:in `<top (required)>'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `block in load'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/engine.rb:652:in `block in load_config_initializer'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/notifications.rb:166:in `instrument'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/engine.rb:651:in `load_config_initializer'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/engine.rb:616:in `block (2 levels) in <class:Engine>'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/engine.rb:615:in `each'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/engine.rb:615:in `block in <class:Engine>'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:30:in `instance_exec'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:30:in `run'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:55:in `block in run_initializers'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:44:in `each'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:44:in `tsort_each_child'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/initializable.rb:54:in `run_initializers'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/application.rb:352:in `initialize!'
/home/git/gitlab/config/environment.rb:5:in `<top (required)>'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `block in require'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/application.rb:328:in `require_environment!'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/railties-4.2.8/lib/rails/application.rb:457:in `block in run_tasks_blocks'
/home/git/gitlab/vendor/bundle/ruby/2.3.0/gems/sprockets-rails-3.2.0/lib/sprockets/rails/task.rb:62:in `block (2 levels) in define'
Tasks: TOP => environment
(See full trace by running task with --trace)
The command '/bin/sh -c bash ${GITLAB_BUILD_DIR}/install.sh' returned a non-zero code: 1
``` 
It seems that the bundle command tries to connect to local redis. The exit Code from yarn was successful so it isn't a yarn problem at all. So one interessting point is which docker version do you use ?

**Comment by solidnerd** [comment]
@carukc Which docker version do you use ?

**Comment by carukc** [comment]
Docker version 17.04.0-ce, build 4845c56

**Comment by solidnerd** [comment]
```

**Comment by carukc** [comment]
Thanks.  I just git cloned and then 'make build'

**Comment by solidnerd** [comment]
Same error. It still fails on asset compiling. I'm a little bit more confused because it works on travis.

**Comment by carukc** [comment]
I've just tried a build on a machine with 16.04 (with 9.1.2).  Same error.

**Comment by solidnerd** [comment]
@carukc  in this issue #1188 . It get solved. I will hopefully introduce it tomorrow.

**Comment by carukc** [comment]
Thank you!

**Comment by solidnerd** [comment]
This got fixed now #1208 . I will be available on the latest tag and later in `9.1.3`. If it's fixed reopen the issue.
